### ISSUE TITLE
Improve support of WCHAR in 1.0

### ISSUE THREAD

**Issue by kleuter** [gabime/spdlog#764]
Current support of wchar is very limited (unlike 0.17 version), there's not much sense of declared wchar support when one can only use formatting string as wchar, not arguments.
Only this works:
```cpp
logger->info(L"This is a text {}", "HELLO");
```
but this not:
```cpp
logger->info(L"This is a text {}", L"HELLO");
```
neither this
```cpp
logger->info(L"This is a text {}", std::wstring(L"HELLO") );
```

**Comment by gabime** [comment]
which version ars you using now? 1.0? or 0.17.0?

**Comment by kleuter** [comment]
I just downloaded the latest from the github v1.x branch.

**Comment by kleuter** [comment]
any progress on this regression bug?

**Comment by gabime** [comment]
Fixed in 8b42b7d269f3397b04e9e4c8f236c4b922de410b
